src/Controls/StreetControls.js

```diff
@@ -77,8 +79,143 @@ class StreetControls extends FirstPersonControls {
         this.previousPosition = newPreviousPosition;
     }
 
+    firstPickedObject(event) {
+        // basic threeJs pick with raycaster
+        const normalized = this.view.viewToNormalizedCoords(event);
+        this.raycaster.setFromCamera(normalized.clone(), this.camera);
+        const intersects = this.raycaster.intersectObjects(this.view.scene.children, true);
+
+        // skipping specific objects:
+        // - skip the two surfaces (rectangle and circle)
+        // - skip all the object without face, because we use the face to manage surface angle, so it's needed.
+        let i = 0;
+        while (i < intersects.length &&
```

Comment:
With picking itowns, you can filter by layer. this should avoid this loop.

Reply:
I don't manage to make it work with existing picking method from iTowns, because I need to pick the TileMesh object, to get the face and the normal. So I prefer using threeJs picking.

Reply:
Finally, I use existing picking method !! :)